src/storage/trie/merkle_trie_tests.rs

```diff
@@ -138,4 +139,219 @@ mod tests {
+                let child_char = xprefix[i];
+                let mut child_hashes = node.child_hashes().clone();
+                child_hashes.insert(child_char, vec![0; 32]);
+                node.set_child_hashes(child_hashes);
+            }
+        }
+        assert_ne!(
+            hex::encode(trie.root_hash().unwrap()),
+            hex::encode(root_hash.clone())
+        );
+
+        // Now, attach the prefix node to the trie
+        let mut txn_batch = RocksDbTransactionBatch::new();
+        let r1 = trie.attach_to_root(ctx, db, &mut txn_batch, &key);
+        db.commit(txn_batch).unwrap();
+        trie.reload(db).unwrap();
+        assert!(r1.is_ok());
+
+        let mut txn_batch = RocksDbTransactionBatch::new();
+        let r = trie.recalculate_hashes(ctx, db, &mut txn_batch, 6);
+        db.commit(txn_batch).unwrap();
+        assert!(r.is_ok());
+
+        let (is_attached, was_created) = r1.unwrap();
+        assert!(is_attached && was_created); // Assert that it was attached
+
+        // Now, the root hashes should match the original
+        assert_eq!(
+            hex::encode(trie.root_hash().unwrap()),
+            hex::encode(root_hash.clone())
+        );
+
+        assert_eq!(trie.items().unwrap(), 4);
+
+        // Attaching it again should be a no-op
+        let mut txn_batch = RocksDbTransactionBatch::new();
+        let r1 = trie.attach_to_root(ctx, db, &mut txn_batch, &key);
+        db.commit(txn_batch).unwrap();
+        assert!(r1.is_ok());
+
+        let mut txn_batch = RocksDbTransactionBatch::new();
+        let r = trie.recalculate_hashes(ctx, db, &mut txn_batch, 6);
+        db.commit(txn_batch).unwrap();
+        assert!(r.is_ok());
+
+        let (is_attached, was_created) = r1.unwrap();
+        assert!(is_attached && !was_created); // Assert that it was not created again
+
+        // root hashes should still match the original
+        assert_eq!(
+            hex::encode(trie.root_hash().unwrap()),
+            hex::encode(root_hash)
+        );
+
+        // Attempting to attach a non-existing node should not be an error, but is_attached and was_created should
+        // both be false
+        let mut txn_batch = RocksDbTransactionBatch::new();
+        let r = trie.attach_to_root(ctx, db, &mut txn_batch, &[0; 16]);
+        db.commit(txn_batch).unwrap();
+
+        assert!(r.is_ok());
+
+        let (is_attached, was_created) = r.unwrap();
+        assert!(!is_attached && !was_created);
+    }
+
+    #[test]
+    fn test_bulk_insert_matches_serial_insert() {
+        let ctx = &Context::new();
+
+        let tmp_path1 = tempfile::tempdir().unwrap();
+        let db1 = &RocksDB::new(tmp_path1.path().to_str().unwrap());
+        db1.open().unwrap();
+
+        let tmp_path2 = tempfile::tempdir().unwrap();
+        let db2 = &RocksDB::new(tmp_path2.path().to_str().unwrap());
+        db2.open().unwrap();
+
+        let branching_factor = 16;
```

Comment:
We should just hardcode this into the trie at this point. We cannot change it anymore without breaking roots.